### PR TITLE
Add `to_uppercase` module and its unit tests

### PR DESCRIPTION
**Pull request by NimahMasuud**
- Created `to_uppercase.py` in the `solutions` directory for converting strings to uppercase.
- Added unit tests for the `to_uppercase` function in `solutions/tests/test_to_uppercase.py`.
- Ensured PEP 8 compliance and code formatting using Ruff.
- All tests are passing and the module is ready for integration.
- Includes examples for handling edge cases like empty strings and non-string inputs.

### CODE REVIEW

File: solutions/tests/test_to_uppercase.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+"""
+
+import unittest
+from solutions.to_uppercase import to_uppercase
+
+
+class TestToUppercase(unittest.TestCase):
+    """Tests for the to_uppercase function."""
+
+    def test_valid_string(self):
```
Comment (KHserv-Creator) [inline solutions/tests/test_to_uppercase.py:17]: Nice work
Test names: The test names could describe more clearly what each line of test has to do.
Feedback:  Instead of test_valid_string you can use  test_with_lower_case for more clarity.

**Review by KHserv-Creator — CHANGES_REQUESTED** [review]
The code structure is clear, and both the function and tests are well-organized. The to_uppercase function handles input validation by raising a TypeError for non-string inputs, which is a good defensive measure. Unit tests cover basic cases well, but more edge cases (like special characters and large strings) could be added.
•	The module-level and function-level docstrings are well-defined, but ensure the author and date fields are completed with accurate information (Created on XX XX XX needs to be filled in).
•	Additionally, you could enhance the function's docstring to mention any edge cases handled explicitly (like empty strings).
However, the code does not pass Ruff and Pylint checks, likely due to issues like line length, unused imports, or docstring formatting. Fixing these will improve readability and maintain compliance with best practices. Since the code does not pass Ruff and Pylint linting checks, it may need further refinement in terms of:
Line lengths: Ensure that lines are not excessively long, keeping them under the standard 80 or 100 characters.
Unused imports: Check for any unused imports or variables that may trigger warnings.
Docstring style: Ensure docstrings follow the recommended format, such as using """ with proper alignment and spacing
Also, please double check the function does not called directly in the function file. 
Feedback: if __name__ == "__main__":
    Overall, the function and tests are solid, but some refinements are needed to meet linting standards and handle more edge cases effectively.  The function and tests are well-organized with clear docstrings and appropriate naming conventions. The TypeError check for non-string inputs is a good defensive measure.  Including doctests in the function docstring is excellent for self-documentation.  Unit tests cover the major cases, including edge cases like an empty string and non-string input.  The function is simple, well-structured, and the unit tests comprehensively validate its behavior. Keep up the good work!

**Comment by NimahMasuud** [comment]
> The code structure is clear, and both the function and tests are well-organized. The to_uppercase function handles input validation by raising a TypeError for non-string inputs, which is a good defensive measure. Unit tests cover basic cases well, but more edge cases (like special characters and large strings) could be added. • The module-level and function-level docstrings are well-defined, but ensure the author and date fields are completed with accurate information (Created on XX XX XX needs to be filled in). • Additionally, you could enhance the function's docstring to mention any edge cases handled explicitly (like empty strings). However, the code does not pass Ruff and Pylint checks, likely due to issues like line length, unused imports, or docstring formatting. Fixing these will improve readability and maintain compliance with best practices. Since the code does not pass Ruff and Pylint linting checks, it may need further refinement in terms of: Line lengths: Ensure that lines are not excessively long, keeping them under the standard 80 or 100 characters. Unused imports: Check for any unused imports or variables that may trigger warnings. Docstring style: Ensure docstrings follow the recommended format, such as using """ with proper alignment and spacing Also, please double check the function does not called directly in the function file. Feedback: if **name** == "**main**": Overall, the function and tests are solid, but some refinements are needed to meet linting standards and handle more edge cases effectively. The function and tests are well-organized with clear docstrings and appropriate naming conventions. The TypeError check for non-string inputs is a good defensive measure. Including doctests in the function docstring is excellent for self-documentation. Unit tests cover the major cases, including edge cases like an empty string and non-string input. The function is simple, well-structured, and the unit tests comprehensively validate its behavior. Keep up the good work!

Changes and Refinements:
	•	Input Validation: The to_uppercase function still handles input validation well by raising a TypeError for non-string inputs.
	•	Edge Cases: Additional edge cases (e.g., special characters and large strings) have been incorporated into both the function’s behavior and unit tests, enhancing robustness.
	•	Docstrings:
	•	The module and function-level docstrings have been updated with accurate author and date fields.
	•	The function’s docstring now explicitly mentions how it handles edge cases like empty strings and special characters.
	•	Linting Compliance: The code now passes both Ruff and Pylint checks. Line lengths, unused imports, and docstring formatting issues have been addressed, ensuring better readability and adherence to best practices.
	•	Testing Improvements:
	•	New unit tests for edge cases have been added.
	•	Test names have been refined for better clarity (e.g., test_valid_string renamed to test_lowercase_string_to_uppercase).
	•	Defensive Programming: Including doctests in the function’s docstring ensures self-documentation and quick validation of examples.

File: solutions/tests/test_to_uppercase.py (hunk 1 above)
Comment (NimahMasuud) [reply solutions/tests/test_to_uppercase.py:17]: Thank you for the feedback! I’ve updated the test names to make them more descriptive and aligned with the purpose of each test. For example, test_valid_string has been renamed to test_lowercase_string_to_uppercase for better clarity. I’ve also reviewed other test names and ensured they accurately reflect the specific case they are testing.

**Review by KHserv-Creator — DISMISSED** [review]

File: solutions/tests/test_to_uppercase.py (hunk 1 above)
Comment (NimahMasuud) [reply solutions/tests/test_to_uppercase.py:17]: > Nice work Test names: The test names could describe more clearly what each line of test has to do. Feedback: Instead of test_valid_string you can use test_with_lower_case for more clarity.

done

**Review by KHserv-Creator — DISMISSED** [review]

**Review by KHserv-Creator — DISMISSED** [review]

**Review by emrebiyik — DISMISSED** [review]

**Review by KHserv-Creator — DISMISSED** [review]

**Review by emrebiyik — APPROVED** [review]
